internal/controller/controllers/agent_controller.go

```diff
@@ -348,6 +358,31 @@ func (r *AgentReconciler) populateEventsURL(log logrus.FieldLogger, agent *aiv1b
 	return nil
 }
 
+func generateControllerLogsDownloadURL(baseURL string, clusterID string, authType auth.AuthType, host string, logsType string) (string, error) {
+	hostID := strfmt.UUID(host)
+	builder := &installer.V2DownloadClusterLogsURL{
+		ClusterID: strfmt.UUID(clusterID),
+		HostID:    &hostID,
+		LogsType:  &logsType,
+	}
+	u, err := builder.Build()
+	if err != nil {
+		return "", err
+	}
+
+	downloadURL := fmt.Sprintf("%s%s", baseURL, u.RequestURI())
+	if authType != auth.TypeLocal {
+		return downloadURL, nil
+	}
+
+	downloadURL, err = gencrypto.SignURL(downloadURL, clusterID, gencrypto.ClusterKey)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to sign cluster controller logs URL")
```

Comment:
```suggestion
		return "", errors.Wrapf(err, "failed to sign %s controller logs URL", logsType)
```

Reply:
Added the change! Thank you for the suggestion :)